Source/Models/APIClient.swift

```diff
@@ -13,15 +17,19 @@ public struct APIClient {
 }
 
 extension APIClient: Client {
-  public func performRequest<T: Request>(request: T, completionHandler: Result<T.ResponseObject, SwishError> -> Void) -> NSURLSessionDataTask {
+  public func performRequest<T: Request>(request: T, scheduler schedule: ((() -> Void) -> Void)? = mainQueueScheduler, completionHandler: Result<T.ResponseObject, SwishError> -> Void) -> NSURLSessionDataTask {
```

Comment:
Do you think the optional is worth it here? It seems like a bit of magic, where consumers could very easily do

```swift
let directScheduler: (() -> Void) -> Void = { $0 }
```

and get the same behavior without adding complexity here or adding semantics to passing `.None`.